### PR TITLE
Scrollable tab bar

### PR DESCRIPTION
**Pull request by bennjii**
* Closes #3

### CODE REVIEW

File: examples/advanced.rs [hunk 1]
```diff
@@ -128,15 +128,38 @@ impl egui_tiles::Behavior<Pane> for TreeBehavior {
...
         _tiles: &egui_tiles::Tiles<Pane>,
         ui: &mut egui::Ui,
-        tile_id: egui_tiles::TileId,
+        _tile_id: egui_tiles::TileId,
+        _tabs: &egui_tiles::Tabs,
+        _offset: f32,
+        _scroll: &mut f32,
+    ) {
+        if ui.button("⏴").clicked() {
+            *_scroll += -45.0;
```
Comment (emilk) [inline examples/advanced.rs:141]: underscore prefix are for unused arguments to silence warnings

```suggestion
        scroll: &mut f32,
    ) {
        if ui.button("⏴").clicked() {
            *scroll += -45.0;
```

File: examples/advanced.rs [hunk 2]
```diff
@@ -128,15 +128,38 @@ impl egui_tiles::Behavior<Pane> for TreeBehavior {
...
+        _tile_id: egui_tiles::TileId,
         _tabs: &egui_tiles::Tabs,
+        _offset: f32,
+        _scroll: &mut f32,
     ) {
-        if ui.button("➕").clicked() {
-            self.add_child_to = Some(tile_id);
+        // if ui.button("➕").clicked() {
+        //     self.add_child_to = Some(tile_id);
+        // }
```
Comment (emilk) [inline examples/advanced.rs:156]: please re-enable

File: examples/advanced.rs [hunk 3]
```diff
@@ -128,15 +128,38 @@ impl egui_tiles::Behavior<Pane> for TreeBehavior {
...
+
+        if ui.button("⏵").clicked() {
+            // Integer value to move scroll by
+            // +'ve is right
+            // -'ve is left
```
Comment (emilk) [inline examples/advanced.rs:161]: plus've?

```suggestion
            // positive is right
            // negative is left
```

File: src/behavior.rs [hunk 4]
```diff
@@ -108,12 +108,27 @@ pub trait Behavior<Pane> {
...
     /// The widgets will be added right-to-left.
-    fn top_bar_rtl_ui(
+    fn top_bar_right_ui(
         &mut self,
         _tiles: &Tiles<Pane>,
         _ui: &mut Ui,
         _tile_id: TileId,
         _tabs: &crate::Tabs,
+        _offset: f32,
+        _scroll: &mut f32,
```
Comment (emilk) [inline src/behavior.rs:118]: Add docstrings explaining what `offset` and `scroll` are, here and below

File: src/container/tabs.rs [hunk 5]
```diff
@@ -15,6 +15,17 @@ pub struct Tabs {
...
 
+#[derive(Default, Clone)]
+struct ScrollState {
+    pub offset: Vec2,
+    pub consumed: Vec2,
+    pub available: Vec2,
+    pub offset_delta: Vec2,
+
+    pub prev_frame_left: bool,
+    pub prev_frame_right: bool,
```
Comment (emilk) [inline src/container/tabs.rs:26]: docstrings!

File: src/container/tabs.rs [hunk 6]
```diff
@@ -108,60 +132,170 @@ impl Tabs {
...
+
+                behavior.top_bar_right_ui(
+                    &tree.tiles,
+                    ui,
+                    tile_id,
+                    self,
+                    scroll_state.offset.x,
+                    &mut scroll_state.offset_delta.x,
+                );
+            }
```
Comment (emilk) [inline src/container/tabs.rs:191]: I don't think this is good - the user should have the option to show buttons in the top-right of the tabs bar no matter what the scroll state is. Something like this:


`[⏴] [tab] [tab] ][⏵] [➕]` where `➕` is always shown, and never scrolled.

The scroll buttons should probably just be hard-coded here in this function

**Review by emilk — CHANGES_REQUESTED** [review]
The scrolling is really nice! But:

I think `top_bar_right_ui` should be reverted to do what is used to do, `top_bar_left_ui` removed, and the `⏴``⏵` buttons added separately (no need to use `Behavior` for it).

Also, the code needs comments.

File: examples/advanced.rs (hunk 3 above)
Comment (bennjii) [reply examples/advanced.rs:161]: Must be a kiwi thing hahaha, changed.

File: src/container/tabs.rs (hunk 6 above)
Comment (bennjii) [reply src/container/tabs.rs:191]: Okay, so just to confirm: You're suggesting we just moved `⏵` and `⏴` into the function, so they are fixed in their behaviour but with `+` exclusively included in `top_bar_right_ui`? Sounds good, it's a better implementation.

**Comment by bennjii** [comment]
Updated it, theres a slight jump when scrolling to the very end as the right menu, I'm tempted to break up the main function `tab_bar_ui` as well, but it seems arduous.

**Review by emilk — APPROVED** [review]

**Pull request merged**